### PR TITLE
feat(components): [autocomplete] add `loading` slot

### PR DESCRIPTION
**Pull request by zhixiaotong**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

## Description

copilot:summary

## Related Issue

Fixes #\_\_\_.

## Explanation of Changes

copilot:walkthrough

### CODE REVIEW

**Comment by zhixiaotong** [comment]
https://github.com/element-plus/element-plus/assets/34681550/183fe5a4-fb94-4a8b-a04e-a5990f8c199b

**Comment by zhixiaotong** [comment]
@sxzz、@btea thx

**Comment by zhixiaotong** [comment]
> 🧪 Playground Preview: https://element-plus.run/?pr=15206 Please comment the example via this playground if needed.

https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXJvdyBjbGFzcz1cImRlbW8tYXV0b2NvbXBsZXRlXCI+XG4gICAgPGVsLWNvbCA6c3Bhbj1cIjEyXCI+XG4gICAgICA8ZGl2IGNsYXNzPVwic3ViLXRpdGxlIG15LTIgdGV4dC1zbSB0ZXh0LWdyYXktNjAwXCI+bG9hZGluZyBpY29uMTwvZGl2PlxuICAgICAgPGVsLWF1dG9jb21wbGV0ZVxuICAgICAgICB2LW1vZGVsPVwic3RhdGVcIlxuICAgICAgICA6ZmV0Y2gtc3VnZ2VzdGlvbnM9XCJxdWVyeVNlYXJjaEFzeW5jXCJcbiAgICAgICAgcGxhY2Vob2xkZXI9XCJQbGVhc2UgaW5wdXRcIlxuICAgICAgICBAc2VsZWN0PVwiaGFuZGxlU2VsZWN0XCJcbiAgICAgID5cbiAgICAgICAgPHRlbXBsYXRlICNsb2FkaW5nPlxuICAgICAgICAgIDxzdmcgY2xhc3M9XCJjaXJjdWxhclwiIHZpZXdCb3g9XCIwIDAgNTAgNTBcIj5cbiAgICAgICAgICAgIDxjaXJjbGUgY2xhc3M9XCJwYXRoXCIgY3g9XCIyNVwiIGN5PVwiMjVcIiByPVwiMjBcIiBmaWxsPVwibm9uZVwiIC8+XG4gICAgICAgICAgPC9zdmc+XG4gICAgICAgIDwvdGVtcGxhdGU+XG4gICAgICA8L2VsLWF1dG9jb21wbGV0ZT5cbiAgICA8L2VsLWNvbD5cbiAgICA8ZWwtY29sIDpzcGFuPVwiMTJcIj5cbiAgICAgIDxkaXYgY2xhc3M9XCJzdWItdGl0bGUgbXktMiB0ZXh0LXNtIHRleHQtZ3JheS02MDBcIj5sb2FkaW5nIGljb24yPC9kaXY+XG4gICAgICA8ZWwtYXV0b2NvbXBsZXRlXG4gICAgICAgIHYtbW9kZWw9XCJzdGF0ZVwiXG4gICAgICAgIDpmZXRjaC1zdWdnZXN0aW9ucz1cInF1ZXJ5U2VhcmNoQXN5bmNcIlxuICAgICAgICBwbGFjZWhvbGRlcj1cIlBsZWFzZSBpbnB1dFwiXG4gICAgICAgIEBzZWxlY3Q9XCJoYW5kbGVTZWxlY3RcIlxuICAgICAgPlxuICAgICAgICA8dGVtcGxhdGUgI2xvYWRpbmc+XG4gICAgICAgICAgPGVsLWljb24gY2xhc3M9XCJpcy1sb2FkaW5nXCI+XG4gICAgICAgICAgICA8c3ZnIGNsYXNzPVwiY2lyY3VsYXJcIiB2aWV3Qm94PVwiMCAwIDIwIDIwXCI+XG4gICAgICAgICAgICAgIDxnXG4gICAgICAgICAgICAgICAgY2xhc3M9XCJwYXRoMiBsb2FkaW5nLXBhdGhcIlxuICAgICAgICAgICAgICAgIHN0cm9rZS13aWR0aD1cIjBcIlxuICAgICAgICAgICAgICAgIHN0eWxlPVwiYW5pbWF0aW9uOiBub25lOyBzdHJva2U6IG5vbmVcIlxuICAgICAgICAgICAgICA+XG4gICAgICAgICAgICAgICAgPGNpcmNsZSByPVwiMy4zNzVcIiBjbGFzcz1cImRvdDFcIiByeD1cIjBcIiByeT1cIjBcIiAvPlxuICAgICAgICAgICAgICAgIDxjaXJjbGUgcj1cIjMuMzc1XCIgY2xhc3M9XCJkb3QyXCIgcng9XCIwXCIgcnk9XCIwXCIgLz5cbiAgICAgICAgICAgICAgICA8Y2lyY2xlIHI9XCIzLjM3NVwiIGNsYXNzPVwiZG90NFwiIHJ4PVwiMFwiIHJ5PVwiMFwiIC8+XG4gICAgICAgICAgICAgICAgPGNpcmNsZSByPVwiMy4zNzVcIiBjbGFzcz1cImRvdDNcIiByeD1cIjBcIiByeT1cIjBcIiAvPlxuICAgICAgICAgICAgICA8L2c+XG4gICAgICAgICAgICA8L3N2Zz5cbiAgICAgICAgICA8L2VsLWljb24+XG4gICAgICAgIDwvdGVtcGxhdGU+XG4gICAgICA8L2VsLWF1dG9jb21wbGV0ZT5cbiAgICA8L2VsLWNvbD5cbiAgPC9lbC1yb3c+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgb25Nb3VudGVkLCByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IHN0YXRlID0gcmVmKCcnKVxuXG5pbnRlcmZhY2UgTGlua0l0ZW0ge1xuICB2YWx1ZTogc3RyaW5nXG4gIGxpbms6IHN0cmluZ1xufVxuXG5jb25zdCBsaW5rcyA9IHJlZjxMaW5rSXRlbVtdPihbXSlcblxuY29uc3QgbG9hZEFsbCA9ICgpID0+IHtcbiAgcmV0dXJuIFtcbiAgICB7IHZhbHVlOiAndnVlJywgbGluazogJ2h0dHBzOi8vZ2l0aHViLmNvbS92dWVqcy92dWUnIH0sXG4gICAgeyB2YWx1ZTogJ2VsZW1lbnQnLCBsaW5rOiAnaHR0cHM6Ly9naXRodWIuY29tL0VsZW1lRkUvZWxlbWVudCcgfSxcbiAgICB7IHZhbHVlOiAnY29va2luZycsIGxpbms6ICdodHRwczovL2dpdGh1Yi5jb20vRWxlbWVGRS9jb29raW5nJyB9LFxuICAgIHsgdmFsdWU6ICdtaW50LXVpJywgbGluazogJ2h0dHBzOi8vZ2l0aHViLmNvbS9FbGVtZUZFL21pbnQtdWknIH0sXG4gICAgeyB2YWx1ZTogJ3Z1ZXgnLCBsaW5rOiAnaHR0cHM6Ly9naXRodWIuY29tL3Z1ZWpzL3Z1ZXgnIH0sXG4gICAgeyB2YWx1ZTogJ3Z1ZS1yb3V0ZXInLCBsaW5rOiAnaHR0cHM6Ly9naXRodWIuY29tL3Z1ZWpzL3Z1ZS1yb3V0ZXInIH0sXG4gICAgeyB2YWx1ZTogJ2JhYmVsJywgbGluazogJ2h0dHBzOi8vZ2l0aHViLmNvbS9iYWJlbC9iYWJlbCcgfSxcbiAgXVxufVxuXG5sZXQgdGltZW91dDogUmV0dXJuVHlwZTx0eXBlb2Ygc2V0VGltZW91dD5cbmNvbnN0IHF1ZXJ5U2VhcmNoQXN5bmMgPSAocXVlcnlTdHJpbmc6IHN0cmluZywgY2I6IChhcmc6IGFueSkgPT4gdm9pZCkgPT4ge1xuICBjb25zdCByZXN1bHRzID0gcXVlcnlTdHJpbmdcbiAgICA/IGxpbmtzLnZhbHVlLmZpbHRlcihjcmVhdGVGaWx0ZXIocXVlcnlTdHJpbmcpKVxuICAgIDogbGlua3MudmFsdWVcblxuICBjbGVhclRpbWVvdXQodGltZW91dClcbiAgdGltZW91dCA9IHNldFRpbWVvdXQoKCkgPT4ge1xuICAgIGNiKHJlc3VsdHMpXG4gIH0sIDUwMDAgKiBNYXRoLnJhbmRvbSgpKVxufVxuY29uc3QgY3JlYXRlRmlsdGVyID0gKHF1ZXJ5U3RyaW5nOiBzdHJpbmcpID0+IHtcbiAgcmV0dXJuIChyZXN0YXVyYW50OiBMaW5rSXRlbSkgPT4ge1xuICAgIHJldHVybiAoXG4gICAgICByZXN0YXVyYW50LnZhbHVlLnRvTG93ZXJDYXNlKCkuaW5kZXhPZihxdWVyeVN0cmluZy50b0xvd2VyQ2FzZSgpKSA9PT0gMFxuICAgIClcbiAgfVxufVxuXG5jb25zdCBoYW5kbGVTZWxlY3QgPSAoaXRlbTogUmVjb3JkPHN0cmluZywgYW55PikgPT4ge1xuICBjb25zb2xlLmxvZyhpdGVtKVxufVxuXG5vbk1vdW50ZWQoKCkgPT4ge1xuICBsaW5rcy52YWx1ZSA9IGxvYWRBbGwoKVxufSlcbjwvc2NyaXB0PlxuXG48c3R5bGU+XG4uY2lyY3VsYXIge1xuICBkaXNwbGF5OiBpbmxpbmU7XG4gIGhlaWdodDogMzBweDtcbiAgd2lkdGg6IDMwcHg7XG4gIGFuaW1hdGlvbjogbG9hZGluZy1yb3RhdGUgMnMgbGluZWFyIGluZmluaXRlO1xufVxuLnBhdGgge1xuICBhbmltYXRpb246IGxvYWRpbmctZGFzaCAxLjVzIGVhc2UtaW4tb3V0IGluZmluaXRlO1xuICBzdHJva2UtZGFzaGFycmF5OiA5MCwgMTUwO1xuICBzdHJva2UtZGFzaG9mZnNldDogMDtcbiAgc3Ryb2tlLXdpZHRoOiAyO1xuICBzdHJva2U6IHZhcigtLWVsLWNvbG9yLXByaW1hcnkpO1xuICBzdHJva2UtbGluZWNhcDogcm91bmQ7XG59XG4ubG9hZGluZy1wYXRoIC5kb3QxIHtcbiAgdHJhbnNmb3JtOiB0cmFuc2xhdGUoMy43NXB4LCAzLjc1cHgpO1xuICBmaWxsOiB2YXIoLS1lbC1jb2xvci1wcmltYXJ5KTtcbiAgYW5pbWF0aW9uOiBjdXN0b20tc3Bpbi1tb3ZlIDFzIGluZmluaXRlIGxpbmVhciBhbHRlcm5hdGU7XG4gIG9wYWNpdHk6IDAuMztcbn1cbi5sb2FkaW5nLXBhdGggLmRvdDIge1xuICB0cmFuc2Zvcm06IHRyYW5zbGF0ZShjYWxjKDEwMCUgLSAzLjc1cHgpLCAzLjc1cHgpO1xuICBmaWxsOiB2YXIoLS1lbC1jb2xvci1wcmltYXJ5KTtcbiAgYW5pbWF0aW9uOiBjdXN0b20tc3Bpbi1tb3ZlIDFzIGluZmluaXRlIGxpbmVhciBhbHRlcm5hdGU7XG4gIG9wYWNpdHk6IDAuMztcbiAgYW5pbWF0aW9uLWRlbGF5OiAwLjRzO1xufVxuLmxvYWRpbmctcGF0aCAuZG90MyB7XG4gIHRyYW5zZm9ybTogdHJhbnNsYXRlKDMuNzVweCwgY2FsYygxMDAlIC0gMy43NXB4KSk7XG4gIGZpbGw6IHZhcigtLWVsLWNvbG9yLXByaW1hcnkpO1xuICBhbmltYXRpb246IGN1c3RvbS1zcGluLW1vdmUgMXMgaW5maW5pdGUgbGluZWFyIGFsdGVybmF0ZTtcbiAgb3BhY2l0eTogMC4zO1xuICBhbmltYXRpb24tZGVsYXk6IDEuMnM7XG59XG4ubG9hZGluZy1wYXRoIC5kb3Q0IHtcbiAgdHJhbnNmb3JtOiB0cmFuc2xhdGUoY2FsYygxMDAlIC0gMy43NXB4KSwgY2FsYygxMDAlIC0gMy43NXB4KSk7XG4gIGZpbGw6IHZhcigtLWVsLWNvbG9yLXByaW1hcnkpO1xuICBhbmltYXRpb246IGN1c3RvbS1zcGluLW1vdmUgMXMgaW5maW5pdGUgbGluZWFyIGFsdGVybmF0ZTtcbiAgb3BhY2l0eTogMC4zO1xuICBhbmltYXRpb24tZGVsYXk6IDAuOHM7XG59XG5Aa2V5ZnJhbWVzIGxvYWRpbmctcm90YXRlIHtcbiAgdG8ge1xuICAgIHRyYW5zZm9ybTogcm90YXRlKDM2MGRlZyk7XG4gIH1cbn1cbkBrZXlmcmFtZXMgbG9hZGluZy1kYXNoIHtcbiAgMCUge1xuICAgIHN0cm9rZS1kYXNoYXJyYXk6IDEsIDIwMDtcbiAgICBzdHJva2UtZGFzaG9mZnNldDogMDtcbiAgfVxuICA1MCUge1xuICAgIHN0cm9rZS1kYXNoYXJyYXk6IDkwLCAxNTA7XG4gICAgc3Ryb2tlLWRhc2hvZmZzZXQ6IC00MHB4O1xuICB9XG4gIDEwMCUge1xuICAgIHN0cm9rZS1kYXNoYXJyYXk6IDkwLCAxNTA7XG4gICAgc3Ryb2tlLWRhc2hvZmZzZXQ6IC0xMjBweDtcbiAgfVxufVxuQGtleWZyYW1lcyBjdXN0b20tc3Bpbi1tb3ZlIHtcbiAgdG8ge1xuICAgIG9wYWNpdHk6IDE7XG4gIH1cbn1cbjwvc3R5bGU+XG4iLCJzcmMvUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTE1MjA2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsInNyYy9lbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTE1MjA2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ1xuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICB9KVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTE1MjA2LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19

File: docs/en-US/component/autocomplete.md [hunk 1]
```diff
@@ -77,13 +87,14 @@ autocomplete/remote-search
...
-| prepend | content to prepend before Input      | -                                        |
-| append  | content to append after Input        | -                                        |
+| Name             | Description                          | Type                                     |
+| ---------------- | ------------------------------------ | ---------------------------------------- |
+| default          | custom content for input suggestions | ^[object]`{ item: Record<string, any> }` |
+| prefix           | content as Input prefix              | -                                        |
+| suffix           | content as Input suffix              | -                                        |
+| prepend          | content to prepend before Input      | -                                        |
+| append           | content to append after Input        | -                                        |
+| loading ^(2.4.5) | override loading icon                | -                                        |
```
Comment (chenxch) [inline docs/en-US/component/autocomplete.md:97]: Would `custom loading content` make more sense?

**Review by chenxch — COMMENTED** [review]
cc @element-plus/backers Is it possible to consider expanding all loading slots uniformly?

File: docs/en-US/component/autocomplete.md (hunk 1 above)
Comment (zhixiaotong) [reply docs/en-US/component/autocomplete.md:97]: > Would `custom loading content` make more sense?

thx. 我们现在需要做一个融合的产品，是从原`angular`部分代码与`vue`代码融合，但视觉需一致。前者是`ng-zorro`，后者是`ep`，两个组件的加载效果不同，我们能解决既有的`loading`效果（通过封装Loading，修改内部的svg元素），但这个`autocomplete`是内部的没法解决，我希望组件库能支持这个效果。

Comment (chenxch) [reply docs/en-US/component/autocomplete.md:97]: `override loading icon` => `custom loading content`

Comment (zhixiaotong) [reply docs/en-US/component/autocomplete.md:97]: 因为`ep`的`loading`本身是支持`svg`的覆盖传入，`v-loading`或全局`loadingInstance`的方案可以满足覆盖的效果。

Comment (zhixiaotong) [reply docs/en-US/component/autocomplete.md]: done

File: docs/en-US/component/autocomplete.md [hunk 2]
```diff
@@ -43,6 +43,16 @@ autocomplete/remote-search
 
 :::
 
+## Custom Loading
```
Comment (btea) [inline docs/en-US/component/autocomplete.md:46]:
```suggestion
## Custom Loading ^(2.5.0)
```

File: docs/en-US/component/autocomplete.md [hunk 3]
```diff
@@ -77,13 +87,14 @@ autocomplete/remote-search
...
-| prepend | content to prepend before Input      | -                                        |
-| append  | content to append after Input        | -                                        |
+| Name             | Description                          | Type                                     |
+| ---------------- | ------------------------------------ | ---------------------------------------- |
+| default          | custom content for input suggestions | ^[object]`{ item: Record<string, any> }` |
+| prefix           | content as Input prefix              | -                                        |
+| suffix           | content as Input suffix              | -                                        |
+| prepend          | content to prepend before Input      | -                                        |
+| append           | content to append after Input        | -                                        |
+| loading ^(2.4.5) | override loading content             | -                                        |
```
Comment (btea) [inline docs/en-US/component/autocomplete.md:97]:
```suggestion
| loading ^(2.5.0) | override loading content             | -                                        |
```

**Comment by btea** [comment]
> cc @element-plus/backers Is it possible to consider expanding all loading slots uniformly?

I think this can make it easier for users to customize extensions better, which I think is great. 👍

**Review by btea — APPROVED** [review]
LGTM

**Review by ryuhangyeong — APPROVED** [review]

**Review by kooriookami — APPROVED** [review]

**Pull request merged**